### ISSUE TITLE
CS 1.6 LAG WHY?? HELP PLEASE

### ISSUE THREAD

**Issue by gibin00** [ValveSoftware/halflife#1836]
Lag cs1.6 
Slow motion and fps_overried off (0) High LAGS!
SYSTEM: WIN 10, 64bit, 3gb ram, 512mb NIVIDA Geforce Grapics CARD, Monitor 60hrz
Net: http://kepkuldes.com/images/3ab51b3bff5131c219a61cf6b419d7c1.png
cs1.6 console: http://kepkuldes.com/images/b18ff501442a1ece1d4f69acc32c130c.jpg
cs1.6 in the game SERVER Net_graph: http://kepkuldes.com/images/01ff412e1e898e290bcaf20924e7d591.jpg
SYSTEM INFORMATION: http://kepkuldes.com/images/8048fcfb66df11ef66c79fdb54f2508a.jpg
cs1.6 launch options:  -nojoy -noforcemaccel -noforcemspd -noforcemparms -32bpp -freq 60 -full -nomaster -nofbo 
and rate 99999 , fps_overried 1, gl_ansio 0, gl_vsync 0 cl_updaterate 999, cl_cmdrate 999
Help please...

**Comment by djdallmann** [comment]
-cl_updaterate is maxed at 102 depending on server configuration which is generally out of your control
-cl_cmdrate has no upper limit but you should probably put it something closer to your fps, general rule of thumb is cl_cmdrate is equal to your fps + 5.
-at the top of your post you say you use fps_override 0 but in system information you say you use it.

**Comment by gibin00** [comment]
Yes fps_overried 0 100.5 fps is fix is lag fps_overried 0 and 1  {0} HIGH LAG 
Not help :(

**Comment by djdallmann** [comment]
CS wasn't designed to have more than 100 fps it breaks the physics and makes you feel slower.

**Comment by gibin00** [comment]
Okey is fps_overried 0 lag how to fix?

**Comment by APGRoboCop** [comment]
I'm not sure but I think the latest HLDS Engine 7559 update on my Linux HLDS server is giving me around 5-20 packet loss. I am not 100% certain but when I used fps_override 1 with fps_max 100 it doesn't seem fix the lag. I did have my net rates set at rate 25000, cl_cmdrate 105 and cl_updaterate 102 in a 16 slot server for DMC and HLDS. I noticed a huge packet loss when I died ingame.

Also I don't quite understand your query nor fully deciphered your texts. Could you maybe explain specifically and maybe in plain English?

**Comment by gibin00** [comment]
Yes ADSL MY Internet.. uh slow motion in game and crosshair slow(maybe lags)

**Comment by gibin00** [comment]
I see, i use fps_override 1

**Comment by Svyatpro** [comment]
Yes, I can confirm too. Something is broken in CS1.6 physics after 2013th updates. When I use fps_max 99.5 I can jump at the mesh edges with zero hp loss like it was on CS 1.6 before updates. When I use fps_max 101 (actual fps 100.5) I cannot jump with zero hp loss. After 2013th updates shooting is not smooth (I mean spraying), it is like it interacts with enemy, if enemy is shooting too it affects my shooting. I don't really know what exactly was changed, but CS 1.6 is now feels really odd, I can this game that way it was before. Please bring back old well-known CS 1.6!!!

**Comment by gibin00** [comment]
How to change? :)

**Comment by shytzedaka** [comment]
I Wonder if Valve even checks any of these issues..

**Comment by In-line** [comment]
Entire goldsrc movement code is broken.

**Comment by ptrBR** [comment]
@mikela-valve

**Comment by eagle-te** [comment]
I experience somewhat the same issue on linux with an i7-8565U(UHD 620), which is faster than my very old Nvidia 8800-GTS 512. That Nvidia ran my game just fine.

I have slowdowns(from 100.5 fps down to min 20-30 fps) everytime I shoot, connection is via LAN and the game starts with https://github.com/FeralInteractive/gamemode .

Playing at 800x600, enabled low quality graphics and disabled HD models in the settings.
I even tried multiple highfps settings userconfig.cfg, to no avail.

@mikela-valve @kisak-valve 
Why is this game, which ran 2003 on an AMD Socket 462(A) **onboard GPU** with 30 fps stable, now such a fps slowdown fest? Even my GeForce4 afterwards, ran this better, than now a UHD 620....

**Comment by SamVanheer** [comment]
Newer hardware probably comes with software drivers for immediate mode so despite being more powerful it isn't actually used at all, it's handled by the CPU only.

To fix that the engine would need an overhaul to use shaders instead so hardware acceleration is possible. That's a pretty major overhaul.

You could try forcing your GPU to be used through the NVIDIA control panel but i don't think that will work if there is no hardware driver support for immediate mode.

**Comment by Luca-Bullentini** [comment]
im guessing is the sound drivers..... will post if I find something

**Comment by demeeek** [comment]
> I experience somewhat the same issue on linux with an i7-8565U(UHD 620), which is faster than my very old Nvidia 8800-GTS 512. That Nvidia ran my game just fine.
> 
> I have slowdowns(from 100.5 fps down to min 20-30 fps) everytime I shoot, connection is via LAN and the game starts with https://github.com/FeralInteractive/gamemode .
> 
> Playing at 800x600, enabled low quality graphics and disabled HD models in the settings.
> I even tried multiple highfps settings userconfig.cfg, to no avail.
> 
> @mikela-valve @kisak-valve
> Why is this game, which ran 2003 on an AMD Socket 462(A) **onboard GPU** with 30 fps stable, now such a fps slowdown fest? Even my GeForce4 afterwards, ran this better, than now a UHD 620....

Try Instaling driverbooster 7.6 there is game mode on my old pc i had signifcanlty better performance my fps was going down to 30fps when i was surrounded with many players cause i played zombie mod  so with driverbooster i had improvemets from 100 do 70 most of the time 100fps .. now i just put 2.5ghz processor i have no problems 100fps doesnt move only with the lag i dont understand what is that in out on the netgraph i hope im going to fix that soon..

**Comment by Solokiller** [comment]
The game is using the immediate mode OpenGL API. It's been obsolete for nearly 2 decades now and new drivers won't always provide good support for it. Older GPUs have drivers that have full support for immediate mode so they have better performance.

**Comment by aBide94** [comment]
I have some freezes on my KZ LAN server, when I have some jumping/climbing things, sometimes I get some lags/freezes but it's about 1-2 seconds., it is not every second, but this makes me nervous. I thought it could be if someone wanted to connect on my server, but however I don't have opened LAN server and I'm not sure, that someone could be to connect to me. 
Maybe I need to fix some settings in CS 1.6 or on LAN server. I tried used some commands, but however I have these f****ing lags.
